### PR TITLE
66E407: news_item_peep_removed

### PR DESCRIPTION
**Pull request by jcdavis**
Not sure if news_item is the right file here, but seems like it.

### CODE REVIEW

File: src/management/news_item.c [hunk 1]
```diff
@@ -344,3 +344,29 @@ void news_item_open_subject(int type, int subject)
...
+                        }
+                }
+                newsItem++;
+        }
+
+        newsItem = RCT2_ADDRESS(0x013CB2D8, rct_news_item);
+        while (newsItem->type != NEWS_ITEM_NULL) {
+                if (type == newsItem->type && sprite_index == newsItem->assoc) {
+                        newsItem->flags |= 0x1;
+                        window_invalidate_by_class(WC_RECENT_NEWS);
```
Comment (jcdavis) [inline src/management/news_item.c]: From my reading, window_invalidate_by_class is the right function when bx is 0? Seems like it from other uses

**Comment by duncanspumpkin** [comment]
Looks good. Yes invalidate_by_class if bx is 0. Generally its invalidate for class if you can't have multiple windows of that type.  I didn't do anything to do with the news functions and i don't quite understand what `0x13CB2D8` is since i see it not used in any other function. @lnz might know the reason. The flag that is being set disables the locate button and news subject so i think this function should probably be renamed disable news or something.

File: src/management/news_item.c (hunk 1 above)
Comment (IntelOrca) [inline src/management/news_item.c]: Yes... basically use by_class for any window that can't have more than one instance. Unless you want to something to apply to every window of a certain type.

**Comment by jcdavis** [comment]
Looks like `0x13CB2D8` is referenced in `66DF32` `news_item_init_queue` as `newsItems[11]` , which makes sense as its `11*268` above the start `0x013CA754`. Seems like a second buffer of some sort - regardless, I'll change to using that since it seems cleaner

**Pull request merged**